tools/flang2/flang2exe/llassem.cpp

```diff
@@ -1239,16 +1239,16 @@ process_dsrt(DSRT *dsrtp, ISZ_T size, char *cptr, bool stop_at_sect,
           fprintf(ASMFIL, ", ");
         if (!i8cnt) {
           ptr = put_next_member(ptr);
-          fprintf(ASMFIL, "zeroinitializer ");
-          free(cptrCopy);
-          return dsrtp;
+          fprintf(ASMFIL, "[ ");
         }
         ptrcnt = 0;
       } else if (!i8cnt) {
         if (!first_data && skip_size)
           fprintf(ASMFIL, ", ");
         ptr = put_next_member(ptr);
-        fprintf(ASMFIL, "[");
+        fprintf(ASMFIL, "zeroinitializer ");
+        free(cptrCopy);
+        return dsrtp;
```

Comment:
I don't follow what changed in the compiler to need to move this block of code around. Do you have an example?

Reply:
Ok, what changed in the compiler:
Step 1: by mistake I placed wrong commit into my pull request for bug #424 
Step 2: my pull request was accepted and landed on master branch
Step 3: it turned out, bug #424 is still not fixed
Step 4: I found out my mistake
Step 5: I opened new pull request with a commit that moves my change to the correct place
Bug #424 comes with minimal reproducer.

Reply:
Thanks!